framework/src/main/java/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -413,24 +416,64 @@ private AnnotationMirror getAnnotationMirrorOfMetaAnnotation(
 
         // Check type-system specific annotations.
         Class<PostconditionAnnotation> metaAnnotation = PostconditionAnnotation.class;
+        Class<String> stringType = String.class;
+        Class<AnnotationMirror> annotationmirrorType = AnnotationMirror.class;
         List<Pair<AnnotationMirror, AnnotationMirror>> declAnnotations =
                 factory.getDeclAnnotationWithMetaAnnotation(methodElement, metaAnnotation);
         for (Pair<AnnotationMirror, AnnotationMirror> r : declAnnotations) {
             AnnotationMirror anno = r.first;
             AnnotationMirror metaAnno = r.second;
-            List<String> expressions =
-                    AnnotationUtils.getElementValueArray(anno, "value", String.class, false);
-            AnnotationMirror postcondAnno = getAnnotationMirrorOfMetaAnnotation(metaAnno, anno);
-            if (postcondAnno == null) {
-                continue;
+
+            @SuppressWarnings("unchecked")
+            List<AnnotationValue> la =
+                    AnnotationUtils.getElementValue(anno, "value", List.class, false);
+
+            {
```

Comment:
This is better in that it does not use exceptions for control flow.  Thanks.

Can you please add some documentation?
For example, I do not understand the distinction between the variables `expressions` and `expressionss`.

Also, please abstract the logic into a new method.
Your new code replaces one or two method calls, and the code will remain clearer if you just change it to call a new method.  That new method may have a complex implementation (40 lines), but it can have a simple and clear specification.

The same comments apply to the code below where you replaced a couple of method calls by a large chunk of code.

Reply:
> When a method has a postcondition specification, the Checker Framework should:
> * establish the postconditions at each call site.  Your tests do not check this.

@mernst I tested the second point and it came out that it is not working on-call site. I am not getting from where this problem is occurring. Please give feedback on it! `@wmdietl` once said that 
>We might need to look into how the StubParser handles repeatable annotations and I suspect some work will be needed there.

Is the problem because of this?

Sample test-
```
import java.util.HashMap;
import java.util.Map;
import org.checkerframework.checker.nullness.qual.EnsuresKeyFor;
import org.checkerframework.checker.nullness.qual.EnsuresKeyForIf;

public class ex {

    Map<String, Integer> map = new HashMap<>();
    Map<String, Integer> map2 = new HashMap<>();

    @EnsuresKeyFor(value = "#1", map = "map")
    @EnsuresKeyFor(value = "#1", map = "map2")
    public void client1(String a, String b, String c) {
        withpostconditionsfunc1(a,b,c);
    }

    @EnsuresKeyFor(value = "#1", map = "map")
    @EnsuresKeyFor(value = "#1", map = "map2")
    public void withpostconditionsfunc1(String a, String b, String c) {
        map.put(a, 1);
        map.put(b, 2);
        map2.put(a, 3);
    }

}
```
Output:-
```
ex.java:13: error: [contracts.postcondition.not.satisfied] the postcondition EnsuresKeyFor about 'a' of this method is not satisfied
    public void client1(String a, String b, String c) {
                ^
1 error
```

Reply:
I have added the documentation. There may be grammatical mistakes. You can help me with that.